udf-examples/src/main/java/com/nvidia/spark/rapids/udf/java/NativeUDFExamplesLoader.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (c) 2021, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.nvidia.spark.rapids.udf.java;
+
+import ai.rapids.cudf.NativeDepsLoader;
+
+import java.io.IOException;
+
+/** Loads the native dependencies for UDF examples with a native implementation */
+public class NativeUDFExamplesLoader {
+  private static volatile boolean isLoaded = false;
+
+  /** Loads native UDF code if necessary */
+  public static void ensureLoaded() {
+    if (!isLoaded) {
```

Comment:
nothing prevents 2 concurrent calls `ensureLoaded` to enter `load()`. Let us not mix volatile and synchronized, and just make `ensureLoaded` synchronized too.

Reply:
Nice catch!  I waffled between two designs on this, and ended in a bad mix of the two. 😞  The intent was to do double-checked locking to avoid the synchronization overhead, as there's potentially many threads trying to do this simultaneously.  But it was missing the second check, and single-checked locking doesn't work. :smile:

However I agree, let's keep it simple for now, we can always optimize if profiles show this is a problem.